Adding a contact immediatly after creating a 1:1 chat crashes the app

```
/home/richard/status/nim-status-client/vendor/nimbus-build-system/vendor/Nim/lib/core/macros.nim(598) onSlotCalled
/home/richard/status/nim-status-client/src/app/profile/view.nim(320) addContact
/home/richard/status/nim-status-client/src/status/contacts.nim(81) addContact
/home/richard/status/nim-status-client/src/status/contacts.nim(49) addContact
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
make: *** [Makefile:439: run-linux] Error 1
```

--- a/src/status/contacts.nim
+++ b/src/status/contacts.nim
@@ -61,7 +61,6 @@ proc addContact*(self: ContactModel, id: string, localNickname: string): string 
     )
 
   let updating = contact.systemTags.contains(":contact/added")
-  echo "C"
   if not updating:
     contact.systemTags.add(":contact/added")
   let nickname =
@@ -73,6 +72,7 @@ proc addContact*(self: ContactModel, id: string, localNickname: string): string 
       localNickname
   result = status_contacts.saveContact(contact.id, contact.ensVerified, contact.ensName, contact.ensVerifiedAt, contact.ensVerificationRetries, contact.alias, contact.identicon, contact.systemTags, nickname)
   self.events.emit("contactAdded", Args())
+  discard requestContactUpdate(contact.id)
   if updating:
     let profile = Profile(
       id: contact.id,

--- a/src/status/contacts.nim
+++ b/src/status/contacts.nim
@@ -1,5 +1,6 @@
 import json, sequtils
 import libstatus/contacts as status_contacts
+import libstatus/accounts as status_accounts
 import profile/profile
 import ../eventemitter
 
@@ -43,8 +44,24 @@ proc getContacts*(self: ContactModel): seq[Profile] =
   self.events.emit("contactUpdate", ContactUpdateArgs(contacts: result))
 
 proc addContact*(self: ContactModel, id: string, localNickname: string): string =
-  let contact = self.getContactByID(id)
+  var contact = self.getContactByID(id)
+  if contact == nil:
+    let alias = status_accounts.generateAlias(id)
+    contact = Profile(
+      id: id,
+      username: alias,
+      identicon: status_accounts.generateIdenticon(id),
+      alias: alias,
+      ensName: "",
+      ensVerified: false,
+      ensVerifiedAt: 0,
+      appearance: 0,
+      ensVerificationRetries: 0,
+      systemTags: @[]
+    )
+
   let updating = contact.systemTags.contains(":contact/added")
+  echo "C"
   if not updating:
     contact.systemTags.add(":contact/added")
   let nickname =

--- a/src/status/libstatus/contacts.nim
+++ b/src/status/libstatus/contacts.nim
@@ -40,6 +40,5 @@ proc saveContact*(id: string, ensVerified: bool, ensName: string, ensVerifiedAt:
     }]
   callPrivateRPC("saveContact".prefix, payload)
 
-proc requestContactUpdate*(): string =
-  let publicKey = $getSetting[string](Setting.PublicKey, "")
+proc requestContactUpdate*(publicKey: string): string =
   callPrivateRPC("sendContactUpdate".prefix, %* [publicKey, "", ""])

--- a/src/status/libstatus/contacts.nim
+++ b/src/status/libstatus/contacts.nim
@@ -1,5 +1,5 @@
 import json
-import core, utils
+import core, utils, types, settings
 from ../profile/profile import Profile
 
 # TODO: remove Profile from here
@@ -39,3 +39,7 @@ proc saveContact*(id: string, ensVerified: bool, ensName: string, ensVerifiedAt:
       "localNickname": localNickname
     }]
   callPrivateRPC("saveContact".prefix, payload)
+
+proc requestContactUpdate*(): string =
+  let publicKey = $getSetting[string](Setting.PublicKey, "")
+  callPrivateRPC("sendContactUpdate".prefix, %* [publicKey, "", ""])

